indexer/src/main/java/au/org/aodn/esindexer/service/IndexerServiceImpl.java

```diff
@@ -257,33 +261,45 @@ public CompletableFuture<ResponseEntity<String>> indexMetadata(String metadataVa
     }
 
     @Override
-    public CompletableFuture<ResponseEntity<String>> indexDataset(Dataset dataset) {
-        try {
+    public List<BulkResponse> indexDataset(String uuid, LocalDate startDate, LocalDate endDate) {
```

Comment:
Without using ComplableFuture, you may get timeout for long running operation, you think it is ok?